### DISCUSSION TITLE
How to save result in .h5?

### DISCUSSION THREAD

**Discussion by kli-jfp** [BLAST-WarpX/warpx#3439] [Installation] [answered]
Hello,
I am running some examples on the WarpX page and things are running smoothly. However, I am having a hard time finding how to save the results in .h5 file format. Maybe I missed a specific build option that needs to be turned on or something that needs to be added to the input decks. Thanks!

**Comment by ax3l** [comment, marked as answer]
Hi @kli-jfp, I am so sorry we overlooked your post.

We implement HDF5 (and ADIOS2) formats through our [openPMD](https://warpx.readthedocs.io/en/22.12/dataanalysis/formats.html) [[1]](https://www.openpmd.org/#/start) backend in the [full diagnostics](https://warpx.readthedocs.io/en/22.12/usage/parameters.html#diagnostics-and-output).

As of WarpX 22.12, you add for your respecive `<diag_name>` the lines
```ini
<diag_name>.format = openpmd
<diag_name>.openpmd_backend = h5  # default is bp for ADIOS output; h5 is for HDF5
```
to the inputs file.

We will make the first line the default in or user-facing examples.

You don't need to add a specifc build option, since we [configure by default](https://warpx.readthedocs.io/en/22.12/install/cmake.html#build-options) with `-DWarpX_OPENPMD=ON`.
